### ISSUE TITLE
Storage space: an upgrade to cargo space

### ISSUE THREAD

**Issue by l29ah** [CleverRaven/Cataclysm-DDA#33287]
**Is your feature request related to a problem? Please describe.**  
I just can't fit enough stuff in my truck, and 250L in a cargo space is unrealistically little (but understandable from the point that stuff cannot be just piled up arbitrarily and still be safe to transport or convenient to access).

**Describe the solution you'd like**  
I suggest a recipe that upgrades `cargo carrier` item with, say, 50 planks to make crates that allows it to store much more items due to vertical stacking.

**Describe alternatives you've considered**  
Copying `hauling space` from Aftershock.

**Additional context**  
![](https://capost.media/upload/iblock/472/472200ef77f2b8f62d70812c12bfd2b5.jpg)

**Comment by kevingranade** [comment]
Your picture contains the answer to your question. The items aren't loose, they're stacked and secured.
The current vehicle storage system assumes that items are loose that they are relatively easily accessible and that they're not piled high enough to cause risk of damaged items on the bottom.
If you wants to extend to the system to soften some of those assumptions and allow for larger cargo capacities there need to be other things in place to limit what can be stored and how they are stored.

**Comment by l29ah** [comment]
> other things in place to limit what can be stored and how they are stored
Can you provide an example?

**Comment by kevingranade** [comment]
For example you could have a system where the volume limit is overridden for a single palletized set of items that occupies an entire tile or a single very large item (cargo container model), or you could have storage that holds more capacity but breaks delicate items placed in it and makes it difficult to find specific items (truck bed model).

**Comment by l29ah** [comment]
Okay, what about a configurable shelving rack like ![](https://images.samsclubresources.com/is/image/samsclub/0007874217941_A?$DT_PDP_Image$)
? You can configure it to accommodate a lot of items of various sizes by inserting an appropriate number of shelves at appropriate spacing, but it cannot be traveled through unlike the current trunks and cargo spaces.

**Comment by kevingranade** [comment]
Sure thing, just implement a system for things falling off the shelves (and potentially breaking and/or falling out of the vehicle) and we can add shelves.
Alternately add it as drawer systems, and handle figuring out which items can fit in your drawers vs how much space you waste, and don't forget that drawer materials are going to limit max capacity and how much volume your drawers take up, etc.

In short, if you have a proposal for adding higher capacity storage, please have a detailed proposal for how exactly it works, and whether it's constructable, and what tradeoffs it makes, because in reality this is a fairly complicated topic and I'm pretty sure there's no good option for, "just make it store more".

**Comment by jeremyshannon** [comment]
This discussion got me thinking. I'm just blue-skying here, but I'd say vehicle tiles that are boardable could maybe have a "clutter" layer. When the tile's storage space is used, instead of items falling through to the pavement, they'd be put on the clutter layer. This would have drawbacks, say for instance 1-19 liters of volume would give increasing penalties to move through the tile, 20-99 liters would block movement so you have to move stuff out of the way, and 100+ liters would also block line-of-sight. On a collision or skid, items in the clutter layer could shift to neighboring tiles with lower clutter volume, with a chance to break fragile items, and a much higher chance to break if they fall out of the vehicle, say off the back of a pickup truck.
 With this kind of setup you'd want to put some large and heavy objects on the back tiles of your pickup, to block the lighter stuff from getting knocked out. Or add a trunk door so you can close it up.
